### PR TITLE
Use ApplicationPartFactory when adding application parts

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/8288

### CODE REVIEW

File: src/Middleware/CORS/test/FunctionalTests/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "devDependencies": {
     "jest": "^23.6.0",
     "merge": "^1.2.1",
-    "puppeteer": "^1.12.2"
+    "puppeteer": "^1.13.0"
```
Comment (javiercn) [inline src/Middleware/CORS/test/FunctionalTests/package.json]: Undo?

**Comment by javiercn** [comment]
@pranavkm When did we break this? 
Do we need to patch this?

**Review by NTaylorMullen — APPROVED** [review]

**Comment by pranavkm** [comment]
> Do we need to patch this?

It's not broken, just doesn't do the right thing all the time. It certainly wouldn't qualify for a patch bug given we have had no user reports and there's a trivial workaround to address it.

**Review by javiercn — APPROVED** [review]

**Comment by pranavkm** [comment]
Test failure investigation is assigned to me (https://github.com/aspnet/AspNetCore-Internal/issues/1913). I'll skip it in a bit.

File: src/Middleware/CORS/test/FunctionalTests/package.json (hunk 1 above)
Comment (pranavkm) [reply src/Middleware/CORS/test/FunctionalTests/package.json]: The build keeps updating this file. I've decided to stop fighting it at this point

**Pull request merged**